eth/backend.go

```diff
@@ -629,6 +632,29 @@ func (s *Ethereum) NetPeerCount() (uint64, error) {
 	return sentryPc, nil
 }
 
+func (s *Ethereum) NodesInfo(limit int) (*remote.NodesInfoReply, error) {
+	if limit == 0 || limit > len(s.sentries) {
+		limit = len(s.sentries)
+	}
+
+	nodes := make([]*prototypes.NodeInfoReply, 0, limit)
+	for i := 0; i < limit; i++ {
+		sc := s.sentries[i]
+
+		nodeInfo, err := sc.NodeInfo(context.Background(), nil)
+		if err != nil {
+			log.Error("sentry nodeInfo", "err", err)
+		}
+
+		nodes = append(nodes, nodeInfo)
+	}
+
+	nodesInfo := &remote.NodesInfoReply{NodesInfo: nodes}
+	nodesInfo.NodesInfo = nodesInfo.NodesInfo[:sortutil.Dedupe(nodesInfo)]
```

Comment:
I looked up `Dedupe` documentation (examples) and it looks like the `nodesInfo` needs to be sorted for it to work correctly. These are examples they give:
```
a := []int{4, 1, 2, 1, 3, 4, 2}
fmt.Println(a[:Dedupe(sort.IntSlice(a))])

b := []string{"foo", "bar", "baz", "bar", "foo", "qux", "qux"}
fmt.Println(b[:Dedupe(sort.StringSlice(b))])
```

I do not see where `nodesInfo` is sorted

Reply:
Actually, all it needs is `sort.Interface`. `Dedupe` function sorts the slice internally and moves all duplicates to the bottom. Let me modify the test a little to show that it works as intended even for unsorted slices.